app/src/componentLibrary/inputNumeric/inputNumeric.jsx

```diff
@@ -0,0 +1,161 @@
+import MinusIcon from './img/minus-inline.svg';
+import PlusIcon from './img/plus-inline.svg';
+import { DEFAULT_WIDTH_CH, KEYCODES_MAP, MAX_WIDTH_CH } from './constants';
+import styles from './inputNumeric.scss';
+
+const cx = classNames.bind(styles);
+
+export const InputNumeric = ({
+  value,
+  placeholder,
+  disabled,
+  onChange,
+  onFocus,
+  onBlur,
+  label,
+  postfix,
+  min,
+  max,
+  title,
+  error,
+  touched,
+}) => {
+  const inputRef = useRef();
+  const handleChange = (event) => {
+    const newValue = event.target.value.replace(/^0(?=\d+|^\d)/g, '');
+    if (newValue === '') {
+      onChange('');
+      return;
+    }
+    if (newValue >= min && newValue <= max) {
+      onChange(+newValue);
+    }
+  };
+  const handleKeyDown = (event) => {
+    const { keyCode } = event;
+
+    if (Object.values(KEYCODES_MAP).includes(keyCode)) {
+      return;
+    }
+    if (keyCode < 48 || keyCode > 57 || event.shiftKey) {
+      event.preventDefault();
+    }
+  };
+  const handleDecrease = () => {
+    const newValue = +value - 1;
+    if (newValue >= min && newValue <= max) {
+      onChange(newValue);
+    }
+  };
+  const handleIncrease = () => {
+    const newValue = +value + 1;
+    if (newValue >= min && newValue <= max) {
+      onChange(newValue);
+    }
+  };
+  const placeholderValue = placeholder + postfix;
+  const inputWidth = useMemo(() => {
+    let width = (String(value) || placeholderValue).length;
+    if (postfix && !value) {
+      width += 1;
+    }
+
+    return width > MAX_WIDTH_CH ? `${MAX_WIDTH_CH}ch` : `${width || DEFAULT_WIDTH_CH}ch`;
+  }, [placeholderValue, postfix, value]);
+  const handleInputFieldClick = () => {
+    if (inputRef && inputRef.current && inputRef.current.focus) {
+      inputRef.current.focus();
+    }
+    onFocus();
+  };
+
+  return (
+    <div className={cx('container', { disabled })}>
+      {label && <div className={cx('label')}>{label}</div>}
+      <div
+        className={cx('input-container', {
+          filled: !!value || value === 0,
+          error,
+          disabled,
+          touched,
+        })}
+        title={title}
+      >
+        <span className={cx('sign', { disabled })} onClick={disabled ? null : handleDecrease}>
+          {Parser(MinusIcon)}
+        </span>
+        <span className={cx('input-field')} onClick={handleInputFieldClick}>
+          <input
+            ref={inputRef}
+            className={cx('input')}
+            type="number"
+            value={value}
+            placeholder={placeholderValue}
+            disabled={disabled}
+            min={min}
+            max={max}
+            onKeyDown={disabled ? null : handleKeyDown}
+            onChange={disabled ? null : handleChange}
+            onFocus={disabled ? null : onFocus}
+            onBlur={disabled ? null : onBlur}
+            style={{ width: inputWidth }}
+          />
+          {!!postfix && (value === 0 || !!value) && <span>{postfix.slice(0, 1)}</span>}
```

Comment:
What is the reason of doing this `postfix.slice(0, 1)`?

Reply:
To prevent entering more than one character